tar_test.go

```diff
@@ -0,0 +1,127 @@
+package mtree
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func ExampleStreamer() {
+	fh, err := os.Open("./testdata/test.tar")
+	if err != nil {
+		// handle error ...
+	}
+	str := NewTarStreamer(fh, nil)
+	if err := extractTar("/tmp/dir", str); err != nil {
+		// handle error ...
+	}
+
+	dh, err := str.Hierarchy()
+	if err != nil {
+		// handle error ...
+	}
+
+	res, err := Check("/tmp/dir/", dh, nil)
+	if err != nil {
+		// handle error ...
+	}
+	if len(res.Failures) > 0 {
+		// handle validation issue ...
+	}
+}
+func extractTar(root string, tr io.Reader) error {
+	return nil
+}
+
+func TestTar(t *testing.T) {
+	/*
+		data, err := makeTarStream()
+		if err != nil {
+			t.Fatal(err)
+		}
+		buf := bytes.NewBuffer(data)
+		str := NewTarStreamer(buf, append(DefaultKeywords, "sha1"))
+	*/
+	/*
+		// open empty folder and check size.
+		fh, err := os.Open("./testdata/empty")
+		if err != nil {
+			t.Fatal(err)
+		}
+		log.Println(fh.Stat())
+		fh.Close() */
+	fh, err := os.Open("./testdata/test.tar")
```

Comment:
for this archive that would be committed to the repo, instead of the whole `./.git/` directory, how about a simpler one that includes a few different things like:
* file(s)
* a couple directories deep
* symlink (valid and a broken one)
* path with a space or such that the vis() will be triggered